packages/integrations/vue/src/index.ts

```diff
@@ -42,18 +49,39 @@ function virtualAppEntrypoint(options?: Options) {
 				return resolvedVirtualModuleId;
 			}
 		},
-		load(id: string) {
+		async load(id: string) {
+			const noop = `export const setup = () => {}`;
 			if (id === resolvedVirtualModuleId) {
-				if (options?.appEntrypoint) {
-					return `export { default as setup } from "${options.appEntrypoint}";`;
+				if (options.appEntrypoint) {
+					try {
+						let resolved;
+						if (options.appEntrypoint.startsWith('.')) {
+							resolved = await this.resolve(fileURLToPath(new URL(options.appEntrypoint, options.root)));
+						} else {
+							resolved = await this.resolve(options.appEntrypoint, fileURLToPath(options.root));
+						}
+						if (!resolved) {
+							throw new Error();
+						}
+						const loaded = await this.load(resolved);
+						if (!loaded.hasDefaultExport) {
+							options.logger.warn(
+								`appEntrypoint \`${options.appEntrypoint}\` does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.`
+							);
+							return noop;
+						}
+						return `export { default as setup } from "${resolved.id}";`;
```

Comment:
Simplified the export statement to a single line. Now, we know that there is a `default` export so this is safe.